pkg/rtc/participant.go

```diff
@@ -1616,11 +1611,6 @@ func (p *ParticipantImpl) onDataMessage(kind livekit.DataPacket_Kind, data []byt
 			u.ParticipantSid = string(p.params.SID)
 			u.ParticipantIdentity = string(p.params.Identity)
 		}
-		if dp.ParticipantIdentity != "" {
```

Comment:
this was always true due to L1604 for non-hidden participants. 
For hidden participants it (wrongly?) assigned the `u.ParticipantIdentity` to the hidden participant.